.github/workflows/k8s-deploy.yml

```
@@ -0,0 +1,350 @@
+        
+        # Check metrics endpoint
+        curl -f http://${SERVICE_IP}:3000/metrics
+
+    - name: Run production smoke tests
+      run: |
+        # Add comprehensive production tests here
+        echo "Running production smoke tests..."
+        
+        # Test main functionality
+        SERVICE_IP=$(kubectl get service walmart-grocery-service -n crewai-team -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+        
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)